mysql-test/suite/galera/t/galera_sequences.test

```diff
@@ -98,3 +92,175 @@ DROP TABLE t1;
 DROP SEQUENCE sq1;
 DROP SEQUENCE sq2;
 SET SESSION wsrep_OSU_method='TOI';
+
+#
+# MDEV-30388 Assertion `!wsrep_has_changes(thd) || (thd->lex->sql_command == SQLCOM_CREATE_TABLE
+# && !thd->is_current_stmt_binlog_format_row()) ||
+# thd->wsrep_cs().transaction().state() == wsrep::transaction::s_aborted' failed
+#
+
+CREATE TABLE t (f INT) engine=innodb;
+LOCK TABLE t WRITE;
+CREATE OR REPLACE SEQUENCE t MAXVALUE=13 INCREMENT BY 1 NOCACHE engine=innodb;
+--error ER_NOT_SUPPORTED_YET
+LOCK TABLE t WRITE;
+INSERT INTO t VALUES (0,0,1,1,1,0,0,0);
+SELECT * from t;
+SELECT NEXTVAL(t);
+UNLOCK TABLES;
+DROP TABLE t;
+
+CREATE SEQUENCE t INCREMENT BY 0 NOCACHE ENGINE=INNODB;
+CREATE TABLE t1(a int not null primary key default nextval(t),
+b int) engine=innodb;
+INSERT INTO t1(b) VALUES (1),(2),(3);
+SELECT * FROM t1;
+
+--connection node_2
+SELECT * FROM t1;
+INSERT INTO t1(b) VALUES (4),(5),(6);
+SELECT * FROM t1;
+
+--connection node_1
+SELECT * FROM t1;
+DROP TABLE t1;
+DROP SEQUENCE t;
+
+#
+# Test Galera SEQUENCE support
+#
+#
+# No MyISAM SEQUENCES
+#
+--error ER_NOT_SUPPORTED_YET
+CREATE SEQUENCE t ENGINE=MYISAM;
+CREATE SEQUENCE t INCREMENT BY 1 NOCACHE ENGINE=INNODB;
+CREATE TABLE t1(a int not null primary key default nextval(t), b int) engine=innodb;
+
+--let $count = 20
+--disable_query_log
+while ($count)
+{
+--connection node_1
+INSERT INTO t1(b) values (1);
+--connection node_2
+INSERT INTO t1(b) values (2);
```

Comment:
what is the expectation here about applying state of queries executed in node_1? if wsrep_sync_wait has bit 4 on, INSERT will be executed only after the INSERT from node_1 has been committed. I suggest to make this explicitly visible by setting wsrep_sync_wait=15 for node_2 and node_1 connections. Or alternatively, if INSERTS should be tested to execute concurrently, add a wait condition to make sure that t1 has been created in node_2

Reply:
Order here does not matter as long as values remain unique across cluster, I will add sync wait to make it clear.